lib/cocoapods/sandbox.rb

```diff
@@ -444,6 +444,27 @@ def local_podspec(name)
       development_pods[root_name]
     end
 
+    # @!group Convenience Methods
+
+    # Writes a file if it does not exist or if its contents have changed.
+    #
+    # @param  [Pathname] path
+    #         The path to read from and write to.
+    #
+    # @param  [String] contents
+    #         The contents to write if they do not match or the file does not exist.
+    #
+    # @return [void]
+    #
+    def self.update_changed_file(path, contents)
+      if path.exist?
+        content_stream = StringIO.new(contents)
+        identical = File.open(path, 'rb') { |f| FileUtils.compare_stream(f, content_stream) }
+        return if identical
+      end
+      File.open(path, 'w') { |f| f.write(contents) }
+    end
```

Comment:
could also use this in `TargetInstallerHelper#update_changed_file`

Reply:
This is actually a bit different as each generator has its own `save_as` method